src/redux/books/books.js

```diff
@@ -1,39 +1,60 @@
-import { v4 as uuidV4 } from 'uuid';
+import bookServices from '../../services/bookServices';
 
+const GET_BOOKS = 'bookstore/book/GET_BOOKS';
 const ADD_BOOK = 'bookstore/book/ADD_BOOK';
 const REMOVE_BOOK = 'bookstore/book/REMOVE_BOOK';
 
-const initialState = [
-  {
-    id: uuidV4(),
-    title: 'Romeo and Juliet',
-    author: 'Will Smith',
-  },
-  {
-    id: uuidV4(),
-    title: 'Things fall apart',
-    author: 'Chinua Achebe',
-  },
-];
+const initialState = [];
 
-export const addBook = (book) => ({
-  type: ADD_BOOK,
-  payload: book,
-});
+export const getBooks = () => async (dispatch) => {
+  try {
+    const result = await bookServices.getFromServer();
+    return dispatch({
+      type: GET_BOOKS,
+      payload: result,
+    });
+  } catch (error) {
+    return error;
+  }
+};
 
-export const removeBook = (id) => ({
-  type: REMOVE_BOOK,
-  payload: id,
-});
+export const addBook = (book) => async (dispatch) => {
+  try {
+    const result = await bookServices.sendToServer(book);
+    return dispatch({
+      type: ADD_BOOK,
+      payload: result,
+    });
+  } catch (error) {
+    return error;
+  }
+};
```

Comment:
- [ ] Currently, when you add a new book it is not displayed on the page. the root cause of this is in this function. the `bookServices.sendToServer` function returns a text "Created" that you are dispatching as the payload of `ADD_BOOK` action. You would not want to do this as it will add the text "Created" as a book in the booksReducer. Instead you might want to simply check the `result === 'Created'` and dispatch an object with the new book details as the payload.

```suggestion
export const addBook = (book) => async (dispatch) => {
  const result = await bookServices.sendToServer(book);
  if (result === 'Created') {
    dispatch({
      type: ADD_BOOK,
      payload: {
        id: book.item_id,
        title: book.title,
        author: book.author,
      },
    });
  }
};
```